slalombuild/secureli
# feat: secureli 248 new relic api url

secureli-248



Closes #248

This adds new logic to prompt the user for a telemetry API URL to send usage statistics to. The URL will default to new relic if a value is not entered. The value is saved to the .secureli.yaml file in order to be committed to source control.


## Changes


* Adding new settings field to telemetry API URL
* Adding prompt logic in order to prompt user for URL
* Updating logic to use either the env var URL or the settings URL
* Updating telemetry env var names to be less generic
* Fix and update existing unit tests, and add unit tests to increase test coverage to 100%

## Testing

* Run `secureli init` in order to be prompted to telemetry url
* Provide url
    * Enter a url to overwrite the default
    * Press enter without entering a value to use the default new relic url
* Expect the correct url to be added in the .secureli.yaml file under `telemetry -> api_url`
* Verify that the URL is used in logging
* Verify that an environment variable URL can be used instead of the settings URL
<img width="677" alt="Screenshot 2024-02-05 at 1 49 08 PM" src="https://github.com/slalombuild/secureli/assets/58826693/89d85faa-70a7-4e00-933e-88ce22d689a9">


## Clean Code Checklist

- [x] Meets acceptance criteria for issue
- [x] New logic is covered with automated tests
- [x] Appropriate exception handling added
- [x] Thoughtful logging included
- [ ] Follow-up work is documented in TODOs
- [ ] TODOs have a ticket associated with them
- [x] No commented-out code included
